doc/content/design/sm-supported-image-formats.md

```diff
@@ -0,0 +1,76 @@
+---
+title: Add supported image formats in sm-list
+layout: default
+design_doc: true
+revision: 2
+status: proposed
+---
+
+# Introduction
+
+At XCP-ng, we are enhancing support for QCOW2 images in SMAPI. The primary
+motivation for this change is to overcome the 2TB size limitation imposed
+by the VHD format. By adding support for QCOW2, a Storage Repository (SR) will
+be able to host disks in VHD and/or QCOW2 formats, depending on the SR type.
+In the future, additional formats—such as VHDx—could also be supported.
+
+We need a mechanism to expose to end users which image formats are supported
+by a given SR. The proposal is to extend the SM API object with a new field
+that clients (such as XenCenter, XenOrchestra, etc.) can use to determine the
+available formats.
+
+# Design Proposal
+
+To expose the available image formats to clients (e.g., XenCenter, XenOrchestra, etc.),
+we propose adding a new field called `supported-image-formats` to the Storage Manager (SM)
+module. This field will be included in the output of the `SM.get_all_records` call.
+
+The `supported-image-formats` field will be populated by retrieving information
+from the SMAPI drivers. Specifically, each driver will update its `DRIVER_INFO`
+dictionary with a new key, `supported_image_formats`, which will contain a list
+of strings representing the supported image formats
+(for example: `["vhd", "raw", "qcow2"]`).
+
+The list designates the driver's preferred VDI format as its first entry. That
+means that when migrating a VDI, the destination storage repository will
+attempt to create a VDI in this preferred format. If the default format cannot
+be used (e.g., due to size limitations), an error will be generated.
```

Comment:
Ideally this needs to be part of the precheck in migration, but I think the idea here is fine